examples/reference-react-forward-ref.js

```diff
@@ -0,0 +1,24 @@
+function enhance(WrappedComponent) {
+  class Enhanced extends React.Component {
+    // ...
+
+    render() {
+      const {forwardedRef, ...rest} = this.props;
+
+      // Assign the custom prop "forwardedRef" as a ref
+      // highlight-range{1-3}
+      return (
+        <WrappedComponent ref={forwardedRef} {...rest} />
+      );
+    }
+  }
+
+  // Intercept the "ref" and pass it as a custom prop, e.g. "forwardedRef"
+  // highlight-range{1-3}
+  function enhanceForwardRef(props, ref) {
```

Comment:
I find this name to be pretty confusing. I'd prefer anonymous functions if we can't give them good names that mean something and aren't just a mashup of other concepts around them.

Reply:
Let's do something like what I did in c41bb29 ?

Reply:
Sounds good to me